src/main.cpp

```diff
@@ -2696,6 +2698,20 @@ bool CheckBlockHeader(const CBlockHeader& block, CValidationState& state, bool f
 
 bool CheckBlock(const CBlock& block, CValidationState& state, bool fCheckPOW, bool fCheckMerkleRoot)
 {
+    // Shortcut checks for almost all early blocks
+    if(chainActive.Height() < SKIP_VALIDATION_HEIGHT || GetBoolArg("-fastsync", false))
```

Comment:
```suggestion
    if(chainActive.Height() < SKIP_VALIDATION_HEIGHT && GetBoolArg("-fastsync", false))
```
I'm not sure what the interface for `GetBoolArg()` is but think we wanted an `and` here?

Reply:
You are absolutely correct there. The code that is encapsulated in the if statement just allows for the validation of the checkpoint blocks if fast sync is enabled. I will fix.

Reply:
Shouldn't it be:
    if(chainActive.Height() < SKIP_VALIDATION_HEIGHT && GetBoolArg("-fastsync", true))

Reply:
No. `/**
 * Return boolean argument or default value
 *
 * @param strArg Argument to get (e.g. "-foo")
 * @param default (true or false)
 * @return command-line argument or default value
 */
bool GetBoolArg(const std::string& strArg, bool fDefault);`

The second argument is a default value if the the flag is not been set.

Reply:
Ok, got it.